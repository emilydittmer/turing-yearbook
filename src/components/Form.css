.form{
  display: grid;
  grid-template-columns: 30% 40% 30%;
}

form {
  grid-column-start: 2;
  background-color: rgba(255, 255, 255, 0.95);
  text-align: center;
}

.form-fields{
  display:flex;
  flex-direction: row;
  justify-content: center;
}

.form-key{
  color: hsl(29, 74%, 59%);
  font-size: 1.25em;
  margin-top: 20px;
  margin-bottom: 5px;
}

input{
  height: 20px;
  margin-top: 20px;
  margin-bottom: 5px;
}

button{
  background-color: hsl(29, 74%, 59%);
  color: #fff;
  height: 50px;
  width: 200px;
  margin-bottom: 10px;
  font-size: 1em;
  font-weight: 700;
}

button:hover{
  background-color: #fff;
  color: hsl(29, 74%, 59%);
  border: 2px solid hsl(29, 74%, 59%);
}