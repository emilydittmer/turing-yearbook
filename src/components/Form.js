import React, { Component } from 'react';
import './Form.css';

class Form extends Component {
  constructor() {
    super();
    this.state = {
      name: '',
      quote: '',
      superlative: '',
      photo: ''
    }
  }
  
  handleValueChange = (event) => {
    this.setState({ [event.target.name] : event.target.value })
  }

  handlePhotoChange = (event) => {
      this.setState({
        photo: URL.createObjectURL(event)
      })
  }

  submitNewStudent = (event) => {
    event.preventDefault();

    const newStudent = {...this.state, id: Date.now()}
    this.props.addNewStudent(newStudent)
    this.resetInputs();
  }

  resetInputs = () => {
    this.setState({ name: '', quote: '', superlative: '', photo: ''})
  }

  render(){
    return(
      <div className='form'>
      <form>
        <div className='form-fields'>
          <h3 className='form-key'>Student Name: </h3>
          <input 
            type= "text"
            placeholder= "Student Name"
            value={this.state.name}
            name="name"
            onChange={event => {this.handleValueChange(event)}}
          />
        </div>
        <div className='form-fields'>
          <h3 className='form-key'>Student Quote: </h3>
          <input 
            type= "text"
            placeholder= "Quote"
            value={this.state.quote}
            name="quote"
            onChange={event => {this.handleValueChange(event)}}
          />
        </div>
        <div className='form-fields'>
          <h3 className='form-key'>Student Superlative: </h3>
          <input 
            type= "text"
            placeholder= "Superlative"
            value={this.state.superlative}
            name="superlative"
            onChange={event => {this.handleValueChange(event)}}
          />
        </div>
        <div className='form-fields'>
          <h3 className='form-key'>Add a photo: </h3>
          <input 
            className="photo-input"
            type="file"
            name="photo"
            onChange={event => {this.handlePhotoChange(event.target.files[0])}}
          />
        </div>
        <br/>
        <div className='form-fields'>
          <button onClick={event => this.submitNewStudent(event)}>Submit</button>
        </div>
      </form>
      </div>
    )
  }
}

export default Form;