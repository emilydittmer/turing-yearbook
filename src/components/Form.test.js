import React from 'react';
import { shallow, mount } from 'enzyme';
import Form from './Form';

describe('Form', () => {
  let wrapper;
  const mockAddNewStudent = jest.fn();
  
  beforeEach(() => {
    wrapper = shallow(<Form 
      addNewStudent = {mockAddNewStudent}
    />)
  })

  it('should match the snapshot with all data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should update state when handleValueChange is called', () => {
    const mockEvent = { target: {name: 'name', value: 'Emily'} }

    wrapper.instance().handleValueChange(mockEvent)

    expect(wrapper.state('name')).toEqual('Emily');
  });


  it('should clear state when resetInputs is called', () => {
    const expected = {name: '', quote: '', superlative: '', photo: ''};
    wrapper.instance().setState({
      name: 'Emily', quote: 'quote', superlative: 'super', photo: 'photo'
    });

    wrapper.instance().resetInputs();

    expect(wrapper.state()).toEqual(expected)
  });

  it('should submit a new student when submitNewStudent is called', () => {

    const spy = jest.spyOn(wrapper.instance(), 'submitNewStudent');
    const mockEvent = { 
      target: { 
        name: 'name', 
        value: 'Emily',
      },
      preventDefault: jest.fn()
    };
    expect(spy).not.toHaveBeenCalled();

    wrapper.instance().submitNewStudent(mockEvent)

    expect(spy).toHaveBeenCalled()
  })
})