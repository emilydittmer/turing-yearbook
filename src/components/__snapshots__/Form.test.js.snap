// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should match the snapshot with all data passed in correctly 1`] = `
<div
  className="form"
>
  <form>
    <div
      className="form-fields"
    >
      <h3
        className="form-key"
      >
        Student Name: 
      </h3>
      <input
        name="name"
        onChange={[Function]}
        placeholder="Student Name"
        type="text"
        value=""
      />
    </div>
    <div
      className="form-fields"
    >
      <h3
        className="form-key"
      >
        Student Quote: 
      </h3>
      <input
        name="quote"
        onChange={[Function]}
        placeholder="Quote"
        type="text"
        value=""
      />
    </div>
    <div
      className="form-fields"
    >
      <h3
        className="form-key"
      >
        Student Superlative: 
      </h3>
      <input
        name="superlative"
        onChange={[Function]}
        placeholder="Superlative"
        type="text"
        value=""
      />
    </div>
    <div
      className="form-fields"
    >
      <h3
        className="form-key"
      >
        Add a photo: 
      </h3>
      <input
        className="photo-input"
        name="photo"
        onChange={[Function]}
        type="file"
      />
    </div>
    <br />
    <div
      className="form-fields"
    >
      <button
        onClick={[Function]}
      >
        Submit
      </button>
    </div>
  </form>
</div>
`;
